DhsTranslator/src/DhsAdapterSim.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * DhsAdapterSim.cpp
+ *
+ *  Created on: Feb 18, 2015
+ *      Author: jluhrs
+ */
+
+#include "defines.h"
+#include "DhsAdapterSim.h"
+#include <ctime>
+#include <locale>
+#include <sstream>
+#include <iomanip>
+
+using namespace std;
+
+DhsAdapterSim::DhsAdapterSim(string &myName, string &serverHost,
+        string &serverName, axutil_log_t * log) {
+    this->myName = myName;
+    this->serverHost = serverHost;
+    this->serverName = serverName;
+    this->nextId = 1;
+    this->log = log;
+
+    pthread_mutex_init(&lock, NULL);
+}
+
+DhsAdapterSim::~DhsAdapterSim() {
+    pthread_mutex_destroy(&lock);
+}
+
+DHS_STATUS DhsAdapterSim::createImage(ImageId &id) {
+    time_t t;
+    time(&t);
+    struct tm *local = localtime(&t);
+    int latchedId;
+
+    pthread_mutex_lock(&lock);
+    {
+        latchedId = nextId;
+        if (++nextId > 10000) {
+            nextId = 1;
+        }
+    }
+    pthread_mutex_unlock(&lock);
+
+
+    stringstream formatedId;
+    formatedId << 'S' << setfill('0') << setw(4) << (local->tm_year + 1900)
+            << setw(2) << local->tm_mon << local->tm_mday << 'S' << setw(4)
+            << nextId;
+
+    id.assign(formatedId.str());
+
+    stringstream msgFormater;
+    msgFormater << "Simulated DHS: Created image " << formatedId.str();
+    AXIS2_LOG_INFO_MSG(log, msgFormater.str().c_str());
+    return DHS_S_SUCCESS;
+}
+
+DHS_STATUS DhsAdapterSim::setImageLifeTime(const ImageId& id,
+        DHS_BD_LIFETIME lifeTime) {
+    stringstream msgFormater;
+    msgFormater << "Simulated DHS: Set lifetime for image " << id << " to "
+            << translateLifetime(lifeTime);
+    AXIS2_LOG_INFO_MSG(log, msgFormater.str().c_str());
+    return DHS_S_SUCCESS;
+}
+
+DHS_STATUS DhsAdapterSim::setImageContrib(const ImageId& id,
+        const vector<string>& contribs) {
+    stringstream msgFormater;
+    msgFormater << "Simulated DHS: Set contributors for image " << id << ": { ";
+
+    for (vector<string>::const_iterator iter = contribs.begin();
```

Comment:
More concise and avoids the entire for loop, but not necessary:
`copy(contribs.begin(), contribs.end(), ostream_iterator(msgFormatter, " "));`

Of course, you would need:
`#include <iterator>
#include <algorithm>`